designer/client/conditions/inline-conditions-absolute-dates.js

```diff
@@ -0,0 +1,215 @@
+            <input className='govuk-input govuk-input--width-2' id='cond-value-minutes' name='cond-value-minutes'
+              type='number' maxLength={2} defaultValue={minutes} required
+              onChange={e => this.updateState({ minutes: e.target.value.padStart(2, '0') })} />
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+class AbsoluteDateValues extends React.Component {
+  constructor (props) {
+    super(props)
+    if (props.value) {
+      const dateComponents = props.value.value.split('-')
+      this.state = {
+        year: dateComponents[0],
+        month: dateComponents[1],
+        day: dateComponents[2]
+      }
+    } else {
+      this.state = {}
+    }
+  }
+
+  updateState (state) {
+    this.setState(state, () => {
+      this.passValueToParentComponentIfComplete()
+    })
+  }
+
+  passValueToParentComponentIfComplete () {
+    const { year, month, day } = this.state
+    if (year && month && day) {
+      this.props.updateValue(new Value(`${year}-${month}-${day}`))
+    }
+  }
+
+  render () {
+    const { year, month, day } = this.state
+
+    return (
+      <div className='govuk-date-input'>
+        <div className='govuk-date-input__item'>
+          <div className='govuk-form-group'>
+            <label htmlFor='cond-value-year' className='govuk-label govuk-label--s'>yyyy</label>
+            <input className='govuk-input govuk-input--width-4' id='cond-value-year' name='cond-value-year'
+              type='number' maxLength={4} minLength={4} defaultValue={year} required
+              onChange={e => this.updateState({ year: e.target.value })} />
+          </div>
+        </div>
+
+        <div className='govuk-date-input__item'>
+          <div className='govuk-form-group'>
+            <label htmlFor='cond-value-month' className='govuk-label govuk-label--s'>MM</label>
+            <input className='govuk-input govuk-input--width-2' id='cond-value-month' name='cond-value-month'
+              type='number' maxLength={2} defaultValue={month} required
+              onChange={e => this.updateState({ month: e.target.value.padStart(2, '0') })} />
+          </div>
+        </div>
+        <div className='govuk-date-input__item'>
+          <div className='govuk-form-group'>
+            <label htmlFor='cond-value-day' className='govuk-label govuk-label--s'>dd</label>
+            <input className='govuk-input govuk-input--width-2' id='cond-value-day' name='cond-value-day'
+              type='number' maxLength={2} defaultValue={day} required
+              onChange={e => this.updateState({ day: e.target.value.padStart(2, '0') })} />
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+class AbsoluteDateTimeValues extends React.Component {
+  constructor (props) {
+    super(props)
+    if (props.value) {
+      const dateTimeComponents = props.value.value.split('T')
+      const timeComponents = dateTimeComponents[1].split(':')
```

Comment:
maybe use moment/momenttz for this?

Reply:
done. Also reminded me of something I'd forgotten to come back and implement so thanks!